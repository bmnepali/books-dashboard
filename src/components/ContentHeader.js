import React from 'react';
import PropTypes from 'prop-types';

const ConentHeader = props => {
  return (
    <section className="page-header">
      <h2>My Books</h2>
      <div className="book-category-filter">
        <div className="dropdown">
          <button
            className="btn btn-link dropdown-toggle btn-shadow btn-rounded"
            type="button"
            id="dropdownMenuButton"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          >
            {props.category || 'All Categories'}
          </button>
          <div className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuButton">
            <a className="dropdown-item" onClick={() => props.changeCategory('Biography')}>
              Biography
            </a>
            <a className="dropdown-item" onClick={() => props.changeCategory('Kids')}>
              Kids
            </a>
            <a className="dropdown-item" onClick={() => props.changeCategory('Sports')}>
              Sports
            </a>
            <a className="dropdown-item" onClick={() => props.changeCategory('Hardcover Fiction')}>
              Fiction
            </a>
            <a className="dropdown-item" onClick={() => props.changeCategory('Drama')}>
              Drama
            </a>
          </div>
        </div>
      </div>
    </section>
  );
};

ConentHeader.propTypes = {
  category: PropTypes.string,
  changeCategory: PropTypes.func,
};

export default ConentHeader;
