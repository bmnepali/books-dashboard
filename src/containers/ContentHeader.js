import { connect } from 'react-redux';

import { changeCategory } from '../actions/BookActions';
import ContentHeader from '../components/ContentHeader';

export const mapDispatchToProps = dispatch => {
  return {
    changeCategory: category => {
      dispatch(changeCategory(category));
    },
  };
};

export const mapStateToProps = store => {
  return {
    category: store.booksStore.category,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ContentHeader);
