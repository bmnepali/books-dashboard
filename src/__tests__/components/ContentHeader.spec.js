import * as React from 'react';

// Load Component
import ContentHeader from '../../components/ContentHeader';

describe('ContentHeader Component', () => {
  describe('rendering testing', () => {
    it('renders ContentHeader component without any error', () => {
      shallow(<ContentHeader />);
    });

    it('matches snapshot', () => {
      const wrapper = shallow(<ContentHeader />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
