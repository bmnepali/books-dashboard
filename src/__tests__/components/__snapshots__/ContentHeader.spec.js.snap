// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader Component rendering testing matches snapshot 1`] = `
<section
  className="page-header"
>
  <h2>
    My Books
  </h2>
  <div
    className="book-category-filter"
  >
    <div
      className="dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-link dropdown-toggle btn-shadow btn-rounded"
        data-toggle="dropdown"
        id="dropdownMenuButton"
        type="button"
      >
        All Categories
      </button>
      <div
        aria-labelledby="dropdownMenuButton"
        className="dropdown-menu dropdown-menu-right"
      >
        <a
          className="dropdown-item"
          onClick={[Function]}
        >
          Biography
        </a>
        <a
          className="dropdown-item"
          onClick={[Function]}
        >
          Kids
        </a>
        <a
          className="dropdown-item"
          onClick={[Function]}
        >
          Sports
        </a>
        <a
          className="dropdown-item"
          onClick={[Function]}
        >
          Fiction
        </a>
        <a
          className="dropdown-item"
          onClick={[Function]}
        >
          Drama
        </a>
      </div>
    </div>
  </div>
</section>
`;
